app/src/main/java/com/framgia/soundclound/data/source/TrackRepository.java

```diff
@@ -1,26 +1,36 @@
 package com.framgia.soundclound.data.source;
 
-import android.support.annotation.NonNull;
-
-import com.framgia.soundclound.data.model.Track;
-
-import java.util.List;
-
 /**
  * Created by Sony on 1/5/2018.
  */
 
-public class TrackRepository implements TrackDataSource {
+public class TrackRepository {
 
-    private TrackDataSource mRemoteDataSource;
+    private static TrackRepository INSTANCE = null;
```

Comment:
WARNING: [SuspiciousConstantFieldName] The field name indicates a constant but its modifiers do not
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#SuspiciousConstantFieldName